lib/rspec/openapi/minitest_hooks.rb

```diff
@@ -13,11 +13,27 @@ def run(*args)
         human_name = name.sub(/^test_/, '').gsub('_', ' ')
         example = Example.new(self, human_name, {}, file_path)
         path = RSpec::OpenAPI.path.then { |p| p.is_a?(Proc) ? p.call(example) : p }
-        record = RSpec::OpenAPI::RecordBuilder.build(self, example: example)
+        record = RSpec::OpenAPI::RecordBuilder.build(self, example: example, extractor: find_extractor)
         RSpec::OpenAPI.path_records[path] << record if record
       end
       result
     end
+
+    def find_extractor
+      names = Bundler.load.specs.map(&:name)
+
+      if names.include?('rails') && defined?(Rails) &&
+         Rails.respond_to?(:application) && Rails.application
+        RSpec::OpenAPI::Extractors::Rails
+      elsif names.include?('hanami') && defined?(Hanami) &&
+            Hanami.respond_to?(:app) && Hanami.app?
+        RSpec::OpenAPI::Extractors::Hanami
+        # elsif defined?(Roda)
+        #   some Roda extractor
+      else
+        RSpec::OpenAPI::Extractors::Rack
+      end
+    end
```

Comment:
Can we extract `find_extractor` from `minitest_hooks.rb` and `rspec_hooks.rb` and deduplicate?